### DISCUSSION TITLE
Can i use it to use side buttons for scrolling pages

### DISCUSSION THREAD

**Discussion by yra-wtag** [noah-nuebling/mac-mouse-fix#174] [Questions] [answered]
In windows i can use my mouse side buttons to scroll webpage or any other pages. It is very accurate and fast. Can i do it with this software. Seems like there is a lot of scrolling related feature but not sure if the one i am looking for is supported.

**Comment by noah-nuebling** [comment]
Dear @yra-wtag, thanks for reaching out!

I think Firefox has that functionality you're looking for built in. Some other mouse-enhancing software also has this feature. I recommend [Steer Mouse](http://plentycom.jp/en/steermouse/).

This is probably not what you're looking for, but if you use the [MMF 2 Beta](https://github.com/noah-nuebling/mac-mouse-fix/releases), there are a few more options. You should be able to assign the side buttons to the page-up and page-down keys. 
But clicking and dragging to scroll  is not a feature in MMF so far. I've got such a feature working in the `version-3` branch, but unfortunately, I don't have time to work on the project and actually get it out there at the moment.

Hope that helped somewhat! Lemme know if you have any more questions.

**Comment by yra-wtag** [comment]
Hi @noah-nuebling Thanks for the answer. Yeah i know Steer Mouse it works very well actually. However i am looking for a open source solution. Basically what i am trying to do is use the extra buttons as scroll wheel. Page up and Page down is not granular enough to facilitate it. Do you have any documentation regarding `version-3` branch then i would be happy to give it a go as you don't have time.

**Reply by noah-nuebling** [reply]
Heyy, sorry for the late reply.

I'm not sure how exactly to get the project running on other systems, since it just works on mine. 
Here's a basic outline for installing `version-3` from memory, but you might have to install additional dependencies and perform additional steps.

- Download Xcode from the App Store
- Clone the Mac Mouse Fix project
  - <img width="920" alt="CleanShot 2022-01-05 at 13 16 39@2x" src="https://user-images.githubusercontent.com/40808343/148283286-54845fe2-9dca-4cae-acc1-eec1c375d15a.png">
- Check out the `version-3` branch 
  - `git checkout version-3`
- Open the `Mouse Fix.xcworkspace` file in Xcode
- Get Xcode to install the Swift Packages that the project depends on 
  - (CocoaLumberjack and swift-log)
- Use CocoaPods to install the remaining dependencies
  - (ReactiveSwift, ReactiveObjc and ReactiveCocoa)
- Change the signing certificate to your Apple ID in Xcode
- Now you should be able to select the "App" or "App - release" build schemes and press the play button to build and run the main app.
- If you need to debug the Helper App standalone, then you'll need to point the "Helper" or "Helper - Release" build schemes to the Helper app bundle embedded in the Main App bundle.

I realize this is a lot and not very detailed. If you wanna go through with building the app yourself and need more info, just let me know. 

I'll try to streamline the process of getting the app to run on other systems at some point.

I can also send you a binary, but it's fairly buggy and incomplete at this point.

**Comment by noah-nuebling** [comment, marked as answer]
Hi! Marking this as answered since I'm moving the topic to Issue #353. Please continue discussion of the feature there. Thanks!
